feature-libs/cart/base/components/added-to-cart-toast/added-to-cart-toast.component.ts

```diff
@@ -0,0 +1,202 @@
+import {
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  HostBinding,
+  OnDestroy,
+  Renderer2,
+} from '@angular/core';
+import { WindowRef } from '@spartacus/core';
+import { Observable, OperatorFunction, Subscription } from 'rxjs';
+import { filter, take } from 'rxjs/operators';
+import {
+  ActiveCartFacade,
+  CartToastItem,
+  CartUiEventAddToCart,
+  CART_TOAST_STATE,
+  OrderEntry,
+} from '@spartacus/cart/base/root';
+import { AddedToCartToastComponentService } from './added-to-cart-toast-component.service';
+
+@Component({
+  selector: 'cx-added-to-cart-toast',
+  templateUrl: './added-to-cart-toast.component.html',
+})
+export class AddedToCartToastComponent implements OnDestroy {
+  protected subscription = new Subscription();
+
+  headerElement: HTMLElement | null;
+
+  toastContainerClass: string;
+
+  timeout: number;
+
+  toastContainerBaseClass = 'toast-container';
+
+  lastToastItem: CartToastItem;
```

Comment:
The goal of handling one toast (because we don't need more anyways) was to simplify the implementation to reflect the simple UI requirement where at no time we need more than one toast to show.

While on the UI we display only one toast at a time, the implementation only add more logic to the existing code, which is the opposite of the simplification we were seeking.

For example, we should consider all the arrays and the loops in the toast handling.  We could start with the component service and then the places where it is called.